### ISSUE TITLE
Scenario tests requiring root certificate fail when run via CI or cross-machine

### ISSUE THREAD

**Issue by roncain** [dotnet/wcf#1105]
When PR #1096 enabled tests to run that are marked with [ConditionalFact(nameof(Root_Certificate_Installed))] they fail from CI because the a root certificate is detected but the trust chain cannot be verified.  Prior to this PR, these tests were skipped unless run self-hosted

Typical failure is:
```
    Tcp_ClientCredentialTypeCertificateCanonicalNameTests.TcpClientCredentialType_Certificate_With_Only_CanonicalName_EchoString(endpointUri: net.tcp://wcfcoresrv2.cloudapp.net/WcfService11//TcpCertificateWithSubjectCanonicalNameLocalhost.svc//tcp-server-subject-cn-localhost-cert, shouldCallSucceed: False) [FAIL]
        System.ServiceModel.Security.SecurityNegotiationException : The X.509 certificate CN=localhost, O=DO_NOT_TRUST, OU=Created by https://github.com/dotnet/wcf; AD436762A2047B67DFBED1E25419EE20A4A929FF chain building failed. The certificate that was used has a trust chain that cannot be verified. Replace the certificate or change the certificateValidationMode. The revocation function was unable to check revocation for the certificate.
         The revocation function was unable to check revocation because the revocation server was offline.
 ```
The tests that fail are:
```
Tcp_ClientCredentialTypeCertificateCanonicalNameTests.TcpClientCredentialType_Certificate_With_Only_CanonicalName_EchoString(endpointUri: net.tcp://wcfcoresrv2.cloudapp.net/WcfService11//TcpCertificateWithSubjectCanonicalNameLocalhost.svc//tcp-server-subject-cn-localhost-cert, shouldCallSucceed: False)
Tcp_ClientCredentialTypeCertificateCanonicalNameTests.TcpClientCredentialType_Certificate_With_Only_CanonicalName_EchoString(endpointUri: net.tcp://wcfcoresrv2.cloudapp.net/WcfService11//TcpCertificateWithSubjectCanonicalNameDomainName.svc//tcp-server-subject-cn-domainname-cert, shouldCallSucceed: False)
Tcp_ClientCredentialTypeCertificateCanonicalNameTests.TcpClientCredentialType_Certificate_With_Only_CanonicalName_EchoString(endpointUri: net.tcp://wcfcoresrv2.cloudapp.net/WcfService11//TcpCertificateWithSubjectCanonicalNameFqdn.svc//tcp-server-subject-cn-fqdn-cert, shouldCallSucceed: True)
Https_ClientCredentialTypeTests.BasicAuthenticationInvalidPwd_throw_MessageSecurityException
Https_ClientCredentialTypeTests.BasicAuthentication_RoundTrips_Echo
Tcp_ClientCredentialTypeTests.SameBinding_DefaultSettings_EchoString
Tcp_ClientCredentialTypeTests.SameBinding_SecurityModeTransport_EchoString
```

**Comment by roncain** [comment]
It appears that these failures are not unique to CI but can happen just doing any cross-machine testing, even with self-host.  My repro steps are:
1. Start the self-hosted WCF service on machine A (I used a Windows Server 2012 R2 VM)
2. On machine B:
    a. Ensure you have the code in PR #1096 (just pull that PR)
    b. Compile and run CertificateGenerator.exe [this step is critical]
    c. Set ServiceUri=name-of-machine-A
    d. Clean and run OuterLoops

You will see all the same "revocation" failures we see in CI.  I was able to get these tests to pass when machine A was a Windows 10 Enterprise VM, but it always fails using Windows Server 2012.

/cc @hongdai @zhenlan @iamjasonp -- this issue blocks checkin in PR #1096 unless we disable these tests.  But if we disable these tests unconditionally, we will have zero coverage for tests using certificates.

**Comment by zhenlan** [comment]
@roncain could you please clarify a little bit the difference of this issue and issue #1108?

**Comment by roncain** [comment]
@Zhenlan -- this issue differs from #1108 because its root cause was failure to open a firewall rule for port 80, causing the CRL validation to fail because it could not contact the CRL service. @hongdai fixed this with PR #1107.  As soon as that was fixed, we got further into the tests and uncovered #1108.

So technically, this current issue is fixed, even though these tests now fail for #1108.  Closing this current issue as 'fixed'.
